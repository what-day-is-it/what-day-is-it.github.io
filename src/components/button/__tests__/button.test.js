import React from 'react';
import { shallow } from 'enzyme';
import Button from '../button';

describe('Button', () => {
  const props = {
    onClick: jest.fn()
  };

  const wrapper = shallow(<Button {...props} />);

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default props', () => {
    expect(wrapper.text()).toEqual('');
    expect(wrapper.find('.dark-blue').exists()).toEqual(true);
  });

  it('should call onClick when clicked', () => {
    wrapper.simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });

  it('should scroll to top and call onClick when clicked if props is set to top', () => {
    wrapper.setProps({ toTop: true });
    wrapper.simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });

  it('should have the disabled style if disabled is true', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper.find('.disabled').exists()).toEqual(true);
  });
});
