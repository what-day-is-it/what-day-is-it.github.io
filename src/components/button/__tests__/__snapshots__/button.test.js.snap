// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button
  className="button regular  dark-blue "
  onClick={[Function]}
  type="button"
/>
`;
